### ISSUE TITLE
Rounded corners on lines and area

### ISSUE THREAD

**Issue by TillitForsikringAS** [highcharts/highcharts#14772]
# Description of the feature
I would love the possibility to add rounded corners to lines and areas in Highchart, like border-radius. Currently working with an iOS app, so I would be really grateful to see this feature there!

Noticed that this feature has been requested before by someone else:
https://www.highcharts.com/forum/viewtopic.php?t=44435

# Library related to the feature
Highcharts

# Proof of Concept/Live example for the feature
![image](https://user-images.githubusercontent.com/44638207/101368143-181a4200-38a7-11eb-82bb-b4dabdc96809.png)

**Comment by pawelfus** [comment]
Hi @TillitForsikringAS 

Thank you for sharing the idea!

PoC: https://jsfiddle.net/BlackLabel/8bcm3ywj/2/
Notes:
- works only with `center` step option
- with a very small change, it's really hard to "round" the connection (e.g. line width = 2px, the difference between two points is ~4px)
- does not work with area charts

**Comment by TillitForsikringAS** [comment]
Thanks @pawelfus, that's great!

Do you consider adding rounded corners for Area/Arearange in the future? 
If so I'd be grateful to be notified.

**Comment by pawelfus** [comment]
If this idea will get enough votes we will consider adding this to `line`, `area` and `arearange` series. All updates will be posted in this GitHub issue.

**Comment by Nishnha** [comment]
> Thank you for sharing the idea!
> 
> PoC: https://jsfiddle.net/BlackLabel/8bcm3ywj/2/ Notes:
> 
> * works only with `center` step option
> * with a very small change, it's really hard to "round" the connection (e.g. line width = 2px, the difference between two points is ~4px)
> * does not work with area charts
> * solution above can be used in Highcharts iOS and Highcharts Android (copy & paste `load` event function into the config)

Hi @pawelfus, I would also like to see this feature added. Our use case is an area chart with the `right` step option, so the PoC does not work for us.

![Image](https://github.com/user-attachments/assets/34892b11-25c2-40d5-aa67-eb939059ace1)

**Comment by Nishnha** [comment]
Thanks @hubertkozik, we tried out your suggestion and it works great!

This is with 5 points for each column.

![Image](https://github.com/user-attachments/assets/53337787-ae46-4694-9f82-74b2299c394f)
![Image](https://github.com/user-attachments/assets/88d58746-5611-473e-90d3-012c3137b270)

I was able to get my desired effect by setting the tickInterval to 5, and hiding the tooltips for the duplicated points.

The dip between the first two columns in the top graph looks a bit awkward. 

![Image](https://github.com/user-attachments/assets/a3e36f8a-d657-4b21-97f5-84d44b9bfd0b)

It might look better if the number of duplicated points is increased when the values between the two columns are relatively close together.

But I think some type of rounded corner feature may be better suited for this.
